adapter/pkg/synchronizer/apis_fetcher.go

```diff
@@ -0,0 +1,202 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Package "synchronizer" contains artifacts relate to fetching APIs and
+ * API related updates from the control plane event-hub.
+ * This file contains functions to retrieve APIs and API updates.
+ */
+
+package synchronizer
+
+import (
+	"archive/zip"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/wso2/product-microgateway/adapter/pkg/auth"
+	logger "github.com/wso2/product-microgateway/adapter/pkg/loggers"
+	"github.com/wso2/product-microgateway/adapter/pkg/tlsutils"
+)
+
+const (
+	apiID        string = "apiId"
+	gatewayLabel string = "gatewayLabel"
+	gwType       string = "type"
+	envoy        string = "Envoy"
+	// Authorization represent the authorization header string.
+	Authorization            string = "Authorization"
+	deploymentDescriptorFile string = "deployments.json"
+	// RuntimeArtifactEndpoint represents the /runtime-artifacts endpoint.
+	RuntimeArtifactEndpoint string = "internal/data/v1/runtime-artifacts"
+)
+
+// FetchAPIs pulls the API artifact calling to the API manager
+// API Manager returns a .zip file as a response and this function
+// returns a byte slice of that ZIP file.
+func FetchAPIs(id *string, gwLabel []string, c chan SyncAPIResponse, serviceURL string,
+	userName string, password string, skipSSL bool, truststoreLocation string,
+	runtimeArtifactEndpoint string, sendType bool) {
```

Comment:
Shall we rename this variable name to resourceEndpoint since this is a general endpoint where any resource can be defined instead of just runtime-artifact endpoint.

Reply:
Fixed